src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,262 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.Scanner;
+
+public class Duke {
+    public static final int MIN_DESCRIPTION_LENGTH = 2;
+    public static final String FILE_PATH = "ip/data/duke.txt";
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        printGreetingMessage();
+        loadTaskFile();
+        boolean isExit = false;
+        Scanner in = new Scanner(System.in);
+        while (!isExit) {
+            try {
+                String line;
+                line = readInput(in);
+                isExit = evaluateInput(line);
+            }catch(DukeException e){
+                printInvalidInputMessage();
+                isExit = false;
+            }
+        }
+    }
+
+    public static void loadTaskFile() {
+        try {
+            loadFileContents(FILE_PATH);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found");
+        } catch (DukeException e) {
+            System.out.println("Invalid File content");
+        }
+    }
+
+    private static void loadFileContents(String filePath) throws FileNotFoundException, DukeException{
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String line  = s.nextLine();
+            if(line.contains("|1|")){
+                addFileTask(line.substring(4));
```

Comment:
Maybe you should avoid redundancy and place addFileTask(line.substring(4)); up. It's being used in both if's... Only if(line.contains("|1|")){ tasks.get(tasks.size() - 1).markAsDone();} makes sense in this case.

Reply:
Also, what does 4 mean? Please, refer to Magic numbers: https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well